sycl/include/sycl/ext/intel/experimental/fpga_annotated_properties.hpp

```diff
@@ -373,6 +335,45 @@ template <typename... Args> struct checkValidFPGAPropertySet {
 
   static constexpr bool value = !(!has_BufferLocation && has_InterfaceConfig);
 };
+template <typename... Args>
```

Comment:
I think this might not be the way we want to do this check, mainly for two reasons:
1. We want to avoid this large trunk of code. It will be hard to maintain if the list get longer and longer.  We should use the template to auto generate this check function in pre-processor
3. This requires the whole list to be checked all the time. When the list is long, this will be quite inefficient. With the template way, the code only checks properties that are given by user code

Reply:
I checked with Yug and it seems we didn't find a way to get the template to work with static_assert. 

The problem is that the pre-processor runs before template processing so you can not get a string customized with template's type and display it on a static_assert.

Let's just create a simple Macro here and makes it a bit cleaner and it should be good to go

Reply:
I saw the error that comes with the macro style. I disagree, we should keep Yug's original version. It is strictly better from a user perspective.

Reply:
I have pushed the macro change. I talked with Abhishek offline and he is okay with it now because the error message does not contain "_key" and is now very similar to what was being printed before.